libnmstate/validator.py

```diff
@@ -92,3 +96,12 @@ def verify_dhcp(state):
                (ip.get('dhcp') or ip.get('autoconf')):
                 logging.warning('%s addresses are ignored when '
                                 'dynamic IP is enabled', family)
+
+            if not ip.get('dhcp') and not ip.get('autoconf'):
```

Comment:
This check is performed before merging with the current state (for an existing iface), therefore, the `dhcp` and `autoconf` may not even be present in the desired state, but existing in the current state.

I am not sure about the implication of moving this verification at a later stage (after the merge).

Reply:
I think `DesiredStateIsNotCurrentError` holds too much errors, providing different exception
might help user to identify whether this failure is caused by human-error(invalid configure settins) or issue of libnm/nmstate.

If we know something will fail for sure due to invalid config, there is no need to waste on apply-and-rollback.